blog/electron-31-0.md

```diff
@@ -0,0 +1,87 @@
+- Removed `WebSQL` support [#41868](https://github.com/electron/electron/pull/41868)
+- `nativeImage.toDataURL` will preserve PNG colorspace [#41610](https://github.com/electron/electron/pull/41610)
+- Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/electron/electron/pull/41432)
+
+### Stack Changes
+
+- Chromium`126.0.6478.36`
+  - [New in 126](https://developer.chrome.com/blog/new-in-chrome-126/)
+  - [New in 125](https://developer.chrome.com/blog/new-in-chrome-125/)
+- Node `20.14.0`
+  - [Node 20.14.0 blog post](https://nodejs.org/en/blog/release/v20.14.0/)
+- V8 `12.6`
+
+Electron 31 upgrades Chromium from `124.0.6367.49` to `126.0.6478.36`, Node from `20.11.1` to `20.14.0`, and V8 from `12.4` to `12.6`.
+
+### New Features
+
+- Added `clearData` method to `Session`. [#40983](https://github.com/electron/electron/pull/40983)
+  - Added options parameter to `Session.clearData` API. [#41355](https://github.com/electron/electron/pull/41355)
+- Added support for Bluetooth ports being requested by service class ID in `navigator.serial`. [#41638](https://github.com/electron/electron/pull/41638)
+- Added support for Node's [`NODE_EXTRA_CA_CERTS`](https://nodejs.org/api/cli.html#node_extra_ca_certsfile) environment variable. [#41689](https://github.com/electron/electron/pull/41689)
+- Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/electron/electron/pull/41432)
+- Implemented support for the web standard [File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API). [#41419](https://github.com/electron/electron/pull/41419)
+- Extended `WebContentsView` to accept pre-existing `WebContents` instances. [#42319](https://github.com/electron/electron/pull/42319)
+- Added a new instance property `navigationHistory` on webContents API with `navigationHistory.getEntryAtIndex` method, enabling applications to retrieve the URL and title of any navigation entry within the browsing history. [#41577](https://github.com/electron/electron/pull/41577) <span style={{fontSize: "small"}}>(Also in [29](https://github.com/electron/electron/pull/41661), [30](https://github.com/electron/electron/pull/41662))</span>
+
+### Breaking Changes
+
+#### Removed: `WebSQL` support
+
+Chromium has removed support for WebSQL upstream, transitioning it to Android only. See
+[Chromium's intent to remove discussion](https://groups.google.com/a/chromium.org/g/blink-dev/c/fWYb6evVA-w/m/pziWcvboAgAJ)
+for more information.
+
+#### Behavior Changed: `nativeImage.toDataURL` will preseve PNG colorspace
+
+PNG decoder implementation has been changed to preserve colorspace data. The
+encoded data returned from this function now matches it.
+
+See [crbug.com/332584706](https://issues.chromium.org/issues/332584706) for more information.
+
+#### Behavior Changed: `window.flashFrame(bool)` will flash dock icon continuously on macOS
```

Comment:
nit: I think the standard way of referencing a BrowserWindow instance in the Electron docs is `win.flashFrame` rather than `window.flashFrame`. This also prevents any confusion with the renderer `window` global.

```suggestion
#### Behavior Changed: `win.flashFrame(bool)` will flash dock icon continuously on macOS
```